hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/DispatchHandleContext.java

```diff
@@ -176,8 +176,21 @@ public AccountID payer() {
     }
 
     @Override
-    public boolean tryToChargePayer(final long amount) {
-        return feeAccumulator.chargeFee(payerId, amount, null).networkFee() == amount;
+    public boolean tryToCharge(@NonNull final AccountID accountId, final long amount) {
+        requireNonNull(accountId);
+        if (amount < 0) {
+            throw new IllegalArgumentException("Cannot charge negative amount " + amount);
+        }
+        return feeAccumulator.chargeFee(accountId, amount, null).networkFee() == amount;
+    }
+
+    @Override
+    public void refundBestEffort(@NonNull final AccountID accountId, final long amount) {
+        requireNonNull(accountId);
+        if (amount < 0) {
+            throw new IllegalArgumentException("Cannot charge negative amount " + amount);
```

Comment:
```suggestion
            throw new IllegalArgumentException("Cannot refund negative amount " + amount);
```

Reply:
Will fix in next PR, ty @Neeharika-Sompalli !